### PR TITLE
Ignore null valued headers #429

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Issue #429.

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
cc @Tratcher @halter73

**Comment by JunTaoLuo** [comment]
Will update WebListener and ServerTest in other PRs

**Comment by halter73** [comment]
:shipit:

**Comment by Tratcher** [comment]
Out of curiosity, how did WebListener and HttpPlatformHandler behave with null or empty header values?

File: test/Microsoft.AspNet.Server.Kestrel.FunctionalTests/ResponseTests.cs [hunk 1]
```diff
@@ -70,5 +73,73 @@ public class ResponseTests
...
+            {
+                var dataset = new TheoryData<string, StringValues, string>();
+
+                // Unknown headers
+                dataset.Add("NullString", (string)null, null);
+                dataset.Add("EmptyString", "", "");
+                dataset.Add("NullStringArray", new string[] { null }, null);
+                dataset.Add("EmptyStringArray", new string[] { "" }, "");
+                dataset.Add("MixedStringArray", new string[] { null, "" }, "");
+                // Unknown headers
```
Comment (Tratcher) [inline test/Microsoft.AspNet.Server.Kestrel.FunctionalTests/ResponseTests.cs]: Did you mean to add tests for known headers here?

**Comment by JunTaoLuo** [comment]
@Tratcher WebListener has a similar issue, headers with null values in a string array will crash though the exception is suppressed. I have not tried with HttpPlatformHandler yet.

**Comment by Tratcher** [comment]
Ok, File a separate bug for WebListener if you're not going to fix it immediately.

**Comment by halter73** [comment]
:shipit:

**Pull request merged**